Kitodo/src/main/java/org/kitodo/production/services/data/ImportService.java

```diff
@@ -1206,4 +1211,19 @@ public Process importProcess(String ppn, int projectId, int templateId, String s
         }
         return tempProcess.getProcess();
     }
+
+    /**
+     * Load doc type metadata keys from provided ruleset.
+     * @param ruleset Ruleset from which doc type metadata keys are loaded and returned
+     * @return list of Strings containing the IDs of the doc type metadata defined in the provided ruleset.
+     * @throws IOException thrown if ruleset file cannot be loaded
+     */
+    public static List<String> getDocTypeMetadata(Ruleset ruleset) throws IOException {
+        RulesetManagementInterface rulesetManagementInterface
+                = ServiceManager.getRulesetManagementService().getRulesetManagement();
+        String rulesetDir = ConfigCore.getParameter(ParameterCore.DIR_RULESETS);
+        String rulesetPath = Paths.get(rulesetDir, ruleset.getFile()).toString();
+        rulesetManagementInterface.load(new File(rulesetPath));
+        return new ArrayList<>(rulesetManagementInterface.getFunctionalKeys(FunctionalMetadata.DOC_TYPE));
```

Comment:
There is no need to copy the result to another list.

    return rulesetManagementInterface.getFunctionalKeys(FunctionalMetadata.DOC_TYPE);

To use along with `contains()` as I mentioned above.